src/sharechain/in_memory.rs

```diff
@@ -341,4 +338,51 @@ impl ShareChain for InMemoryShareChain {
             .cloned()
             .collect())
     }
+
+    async fn hash_rate(&self) -> ShareChainResult<BigUint> {
+        let block_levels = self.block_levels.read().await;
+        if block_levels.is_empty() {
+            return Ok(BigUint::zero());
+        }
+
+        let blocks = block_levels
+            .iter()
+            .flat_map(|level| level.blocks.clone())
+            .sorted_by(|block1, block2| block1.timestamp().cmp(&block2.timestamp()))
+            .tail(BLOCKS_WINDOW);
+
+        // calculate average block time
+        let blocks = blocks.collect_vec();
+        let mut block_times_sum = 0;
+        let mut block_times_count: u64 = 0;
+        for i in 0..blocks.len() {
```

Comment:
would it work to only take the first and last block to get the time passed, divided by amount of blocks?

Reply:
I don't think so as here I'm interested in the average time between 2 blocks, I feel like its maybe more precise